site/app/libraries/database/DatabaseQueries.php

```diff
@@ -6559,6 +6559,11 @@ public function getTimeJoinedQueue($user_id, $queue_code) {
         return $this->course_db->rows()[0]['max'];
     }
 
+    public function getStarType($user_id, $queue_code) {
+        $this->course_db->query("SELECT * FROM queue WHERE user_id = ? AND UPPER(TRIM(queue_code)) = UPPER(TRIM(?)) ", [$user_id, $queue_code]);
+        return $this->course_db->rows()[0]['star_type'];
+    }
+
```

Comment:
I think it only selects the first row of `star_type`...  Could be fixed with `ORDER BY entry_id DESC` or selecting specific `entry_id`.  There is a related bug, more in the comment below.

Can it only select `star_type` instead of a wildcard?

```suggestion
    public function getStarType($user_id, $queue_code, $entry_id) {
        $this->course_db->query("SELECT * FROM queue WHERE user_id = ? AND UPPER(TRIM(queue_code)) = UPPER(TRIM(?)) AND entry_id = ?", [$user_id, $queue_code, $entry_id]);
        return $this->course_db->rows()[0]['star_type'];
    }

```